addons/sale_gelato/models/sale_order.py

```diff
@@ -71,11 +71,33 @@ def action_confirm(self):
         for order in self.filtered(
             lambda o: any(o.order_line.product_id.mapped('gelato_product_uid'))
         ):
+            if message := order._ensure_partner_address_is_complete():
+                raise ValidationError(message)
             order._create_order_on_gelato()
         return res
 
     # === BUSINESS METHODS === #
 
+    def _ensure_partner_address_is_complete(self):
+        """ Ensure that all order's partner address fields required by Gelato are set.
```

Comment:
nit: since you are touching that, we can clean it
```suggestion
        """Ensure that all order's partner address fields required by Gelato are set.
```